governance/program/src/state.rs

```diff
@@ -0,0 +1,225 @@
+    /// Voting threshold in % required to tip the vote
+    /// It's the percentage of tokens out of the entire pool of governance tokens eligible to vote
+    pub vote_threshold: u8,
+
+    /// Minimum % of tokens for a governance token owner to be able to create a proposal
+    /// It's the percentage of tokens out of the entire pool of governance tokens eligible to vote
+    pub token_threshold_to_create_proposal: u8,
+
+    /// Minimum waiting time in slots for a transaction to be executed after proposal is voted on
+    pub min_transaction_hold_up_time: u64,
+
+    /// Governance mint
+    pub governance_mint: Pubkey,
+
+    /// Council mint
+    pub council_mint: Option<Pubkey>,
+
+    /// Program ID that is governed by this Governance
+    pub program: Pubkey,
+
+    /// Time limit in slots for proposal to be open for voting
+    pub max_voting_time: u64,
+
+    /// Running count of proposals
+    pub proposal_count: u32,
+}
+
+/// Governance Proposal
+#[derive(Clone)]
+pub struct Proposal {
+    /// Governance account type
+    pub account_type: GovernanceAccountType,
+
+    /// Governance account the Proposal belongs to
+    pub governance: Pubkey,
+
+    /// Proposal State account
+    pub state: Pubkey,
+
+    /// Mint that creates signatory tokens of this Proposal
+    /// If there are outstanding signatory tokens, then cannot leave draft state. Signatories must burn tokens (ie agree
+    /// to move instruction to voting state) and bring mint to net 0 tokens outstanding. Each signatory gets 1 (serves as flag)
+    pub signatory_mint: Pubkey,
+
+    /// Admin ownership mint. One token is minted, can be used to grant admin status to a new person.
+    pub admin_mint: Pubkey,
+
+    /// Source Token Holding account
+    pub source_holding: Pubkey,
+
+    /// Source Mint - either governance or council mint from Governance
+    pub source_mint: Pubkey,
+}
+
+/// Proposal state
+#[derive(Clone)]
+pub struct ProposalState {
+    /// Governance account type
+    pub account_type: GovernanceAccountType,
+
+    /// Proposal account
+    pub proposal: Pubkey,
+
+    /// Current status of the proposal
+    pub status: ProposalStateStatus,
+
+    /// Total signatory tokens minted, for use comparing to supply remaining during draft period
+    pub total_signing_tokens_minted: u64,
+
+    /// Link to proposal's description
+    pub description_link: [u8; MAX_PROPOSAL_DESCRIPTION_LINK_LENGTH],
+
+    /// Proposal name
+    pub name: [u8; MAX_PROPOSAL_NAME_LENGTH],
+
+    /// When the Proposal ended voting - this will also be when the set was defeated or began executing naturally.
+    pub voting_ended_at: u64,
+
+    /// When the Proposal began voting
+    pub voting_began_at: u64,
+
+    /// when the Proposal entered draft state
+    pub created_at: u64,
+
+    /// when the Proposal entered completed state, also when execution ended naturally.
+    pub completed_at: u64,
+
+    /// when the Proposal entered deleted state
+    pub deleted_at: u64,
```

Comment:
```suggestion
    /// When the Proposal ended voting - this will also be when the set was defeated or began executing naturally.
    pub voting_ended_at: Option<Slot>,

    /// When the Proposal began voting
    pub voting_began_at: Option<Slot>,

    /// when the Proposal entered draft state
    pub created_at: Option<Slot>,

    /// when the Proposal entered completed state, also when execution ended naturally.
    pub completed_at: Option<Slot>,

    /// when the Proposal entered deleted state
    pub deleted_at: Option<Slot>,
```